src/Glpi/Form/Destination/AbstractCommonITILFormDestination.php

```diff
@@ -165,6 +187,13 @@ private function applyPredefinedTemplateFields(array $input): array
         $template = $itil->getITILTemplateToUse(
             entities_id: $_SESSION["glpiactive_entity"]
         );
+        $template_foreign_key = $template::getForeignKeyField();
+
+        if (isset($input[$template_foreign_key])) {
+            $template->getFromDB($input[$template_foreign_key]);
+        } else {
+            $input[$template_foreign_key] = $template->getID();
+        }
```

Comment:
I had to read twice the code to understand the if/else logic.

Maybe this would be a bit more clear, and it would also prevent to get a template from `getITILTemplateToUse()` and override it 5 lines later.

```php
$template_class = $itil::getTemplateClass();
$template_foreign_key = $template::getForeignKeyField();

if (isset($input[$template_foreign_key])) {
    $template = $template_class::getById($input[$template_foreign_key]);
} else {
    $template = $itil->getITILTemplateToUse(
        entities_id: $_SESSION["glpiactive_entity"]
    );
    $input[$template_foreign_key] = $template->getID();
}